client/app/bundles/course/users/components/buttons/UserManagementButtons.tsx

```diff
@@ -70,29 +71,32 @@ const UserManagementButtons: FC<Props> = (props) => {
   };
 
   const onDelete = (): Promise<void> => {
+    setIsDeleting(true);
     return dispatch(deleteUser(user.id))
       .then(() => {
         toast.success(intl.formatMessage(translations.deletionSuccess));
       })
       .catch((error) => {
         toast.error(intl.formatMessage(translations.deletionFailure));
         throw error;
-      });
+      })
+      .finally(() => setIsDeleting(false));
   };
 
   const managementButtons = (
     <div style={{ whiteSpace: 'nowrap' }} key={`buttons-${user.id}`}>
       <SaveButton
         tooltip="Save Changes"
         className={`user-save-${user.id}`}
-        disabled={isSaving}
+        disabled={isSaving || isDeleting}
         onClick={(): Promise<void> => onSave(user)}
         sx={styles.buttonStyle}
       />
       <DeleteButton
         tooltip="Delete User"
         className={`user-delete-${user.id}`}
-        disabled={isSaving}
+        disabled={isSaving || isDeleting}
+        loading={isDeleting}
```

Comment:
Could you help adding this to other <DeleteButton /> components that are changed in this PR as well?